packages/flutter_tools/lib/src/flutter_plugins.dart

```diff
@@ -1137,127 +1137,176 @@ List<PluginInterfaceResolution> resolvePlatformImplementation(
+///   * The [plugin] neither provides an implementation for [implementsPluginName]
+///     nor references a default implementation [defaultImplPluginName].
+(String? implementsPluginName, String? defaultImplPluginName) _getPluginImplCandidateAndDefaultPlugin(Plugin plugin, String platformKey) {
+  String? defaultImplPluginName = plugin.defaultPackagePlatforms[platformKey];
+  if (plugin.platforms[platformKey] == null && defaultImplPluginName == null) {
+    // The plugin doesn't implement this platform nor reference a default implementation.
+    return (null, null);
+  }
+  String? implementsPackage = plugin.implementsPackage;
+  final bool hasInlineDartImplementation =
+      plugin.pluginDartClassPlatforms[platformKey] != null &&
+          plugin.pluginDartClassPlatforms[platformKey] != 'none';
+  if (implementsPackage == null || implementsPackage.isEmpty) {
+    // Plugin does not serve as implementation for another app-facing package.
+    if (defaultImplPluginName == null && !hasInlineDartImplementation) {
+      // Skip native inline PluginPlatform implementation.
+      return (null, null);
+    }
+    if (defaultImplPluginName != null) {
+      return (null, defaultImplPluginName);
+    } else {
+      // An app-facing package (i.e., one with no 'implements') with an
+      // inline implementation should be its own default implementation.
+      // Desktop platforms originally did not work that way, and enabling
+      // it unconditionally would break existing published plugins, so
+      // only treat it as such if either:
+      // - the platform is not desktop, or
+      // - the plugin requires at least Flutter 2.11 (when this opt-in logic
+      //   was added), so that existing plugins continue to work.
+      // See https://github.com/flutter/flutter/issues/87862 for details.
+      final bool isDesktop = platformKey == 'linux' || platformKey == 'macos' || platformKey == 'windows';
+      final semver.VersionConstraint? flutterConstraint = plugin.flutterConstraint;
+      final semver.Version? minFlutterVersion = flutterConstraint != null &&
+          flutterConstraint is semver.VersionRange ? flutterConstraint.min : null;
+      final bool hasMinVersionForImplementsRequirement = minFlutterVersion != null &&
+          minFlutterVersion.compareTo(semver.Version(2, 11, 0)) >= 0;
+      if (!isDesktop || hasMinVersionForImplementsRequirement) {
+        // Dart plugin serves as its own default implementation.
+        implementsPackage = plugin.name;
+        defaultImplPluginName = plugin.name;
+        return (implementsPackage, defaultImplPluginName);
+      } else {
+        // If the inline dart implementation is desktop and it's flutter min
+        // constraint isn't fulfilled, it isn't eligible for auto-registration.
+        return (null, null);
+      }
+    }
+  }
+
+  if (!hasInlineDartImplementation) {
+    // If there's no Dart implementation, there's nothing to register.
```

Comment:
This comment seems out of place now, because Dart registration isn't part of the scope of this extracted function.

Reply:
This method still only handles Dart plugins up to now. What alternative do you have in mind?

Reply:
The comment is now obsolete due to different handling in 1c0aff6042587695feae7d87539f05e31fbcb56a